File: .sdk-repos/orchestration-cluster-api-js/examples/exporting.ts
// Compilable usage examples for exporting operations.
// These examples are type-checked during build to guard against API regressions.

import { createCamundaClient } from '@camunda8/orchestration-cluster-api';

//#region PauseExporting
async function pauseExportingExample() {
  const camunda = createCamundaClient();

  // With `soft: true` exporting keeps running but its position is not committed,
  // so the log is still not compacted — use it when exporting must keep
  // progressing, for example while a backup is taken.
  await camunda.pauseExporting({ soft: true });
}
//#endregion PauseExporting

//#region ResumeExporting
async function resumeExportingExample() {
  const camunda = createCamundaClient();

  await camunda.resumeExporting();
}
//#endregion ResumeExporting

//#region GetExportingStatus
async function getExportingStatusExample() {
  const camunda = createCamundaClient();

  // Reports the aggregated exporting status of the physical tenant — useful to
  // confirm exporting has actually paused before taking a backup, and that it
  // has resumed afterwards.
  const { status } = await camunda.getExportingStatus();
  console.log(`Exporting status: ${status}`);
}
//#endregion GetExportingStatus

//#region PauseClusterExporting
async function pauseClusterExportingExample() {
  const camunda = createCamundaClient();

  // Cluster-admin variant: pauses exporting on every physical tenant of the
  // cluster. With `soft: true` exporting keeps running but its position is not
  // committed, so the log is still not compacted.
  await camunda.pauseClusterExporting({ soft: true });
}
//#endregion PauseClusterExporting

//#region ResumeClusterExporting
async function resumeClusterExportingExample() {
  const camunda = createCamundaClient();

  await camunda.resumeClusterExporting();
}
//#endregion ResumeClusterExporting

//#region GetClusterExportingStatus
async function getClusterExportingStatusExample() {
  const camunda = createCamundaClient();

  // Reports the aggregated exporting status of the whole cluster — useful to
  // confirm exporting has paused everywhere before taking a cluster-wide backup.
  const { status } = await camunda.getClusterExportingStatus();
  console.log(`Cluster exporting status: ${status}`);
}
//#endregion GetClusterExportingStatus

// Suppress "declared but never read"
void pauseExportingExample;
void resumeExportingExample;
void getExportingStatusExample;
void pauseClusterExportingExample;
void resumeClusterExportingExample;
void getClusterExportingStatusExample;

File: .sdk-repos/orchestration-cluster-api-python/examples/exporting.py
# Compilable usage examples for exporting operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import CamundaClient


# region PauseExporting
def pause_exporting_example() -> None:
    client = CamundaClient()

    # With `soft=True` exporting keeps running but its position is not committed,
    # so the log is still not compacted. Use it when exporting must keep
    # progressing -- for example while a backup is taken.
    client.pause_exporting(soft=True)
# endregion PauseExporting


# region ResumeExporting
def resume_exporting_example() -> None:
    client = CamundaClient()

    client.resume_exporting()
# endregion ResumeExporting


# region GetExportingStatus
def get_exporting_status_example() -> None:
    client = CamundaClient()

    result = client.get_exporting_status()

    # The status is aggregated over every replica of every partition, so `MIXED`
    # means a pause or resume is still in flight or was only partially applied.
    # Only `PAUSED` and `SOFT_PAUSED` confirm that exporting has stopped.
    print(f"Status: {result.status}")
# endregion GetExportingStatus


# region GetClusterExportingStatus
def get_cluster_exporting_status_example() -> None:
    client = CamundaClient()

    # Requires the cluster-admin security chain — not the Orchestration Cluster
    # user credentials. Only `PAUSED` and `SOFT_PAUSED` confirm a cluster-wide
    # pause; any other value means at least one physical tenant is still active.
    result = client.get_cluster_exporting_status()

    print(f"Cluster exporting status: {result.status}")
# endregion GetClusterExportingStatus


# region PauseClusterExporting
def pause_cluster_exporting_example() -> None:
    client = CamundaClient()

    # Pauses exporting on every physical tenant in one call.
    # With `soft=True` the position is not committed, so the log is not compacted,
    # which is the right mode for taking a consistent backup without stopping
    # real processing work.
    client.pause_cluster_exporting(soft=True)
# endregion PauseClusterExporting


# region ResumeClusterExporting
def resume_cluster_exporting_example() -> None:
    client = CamundaClient()

    # Resumes exporting on every physical tenant after a pause or soft pause.
    client.resume_cluster_exporting()
# endregion ResumeClusterExporting
